src/nixpacks/app.rs

```diff
@@ -23,13 +23,7 @@ impl App {
     }
 
     pub fn includes_file(&self, name: &str) -> bool {
-        for path in &self.paths {
-            if path.file_name().unwrap() == name {
-                return true;
-            }
-        }
-
-        false
+        fs::read_to_string(self.source.join(name)).is_ok()
```

Comment:
Instead of reading the entire file into memory, I think it would be better to use something like [`fs::canonicalize`](https://doc.rust-lang.org/std/fs/fn.canonicalize.html) to simple check if the file exists on disk.

![image](https://user-images.githubusercontent.com/3044853/161569153-302f5145-9b14-4231-af1d-df610eeb78f7.png)

Something like

```rust
    pub fn includes_file(&self, name: &str) -> bool {
        let path = self.source.join(name);
        path.canonicalize().is_ok()
    }
```